### PR TITLE
KIN-219 Add delete category functionality without files management

### PR DESCRIPTION
**Pull request by Cristianpan**
[KIN-219](https://loopcrack.atlassian.net/browse/KIN-219) Small [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=loopcrack-org&utm_campaign=badge_info)

### Problem

The admin user needs to delete a category so that in case there is no longer needed it can discarded by the system

### Solution

Implement the delete category functionality

### Requirements checklist

Include a list of requirements necessary to consider this pull request complete

- [x] It must delete the register from the database

For the moment, the files inputs will be ignored and when storing on the database, it will be a hardcoded value.
NOTES: The files functionality will be integrated late
### Testing

Manual testing
- [x] Select a category and delete it from the administration panel and check that the category data has been removed from the database along with its associated tags and files.

## Demo

[KIN-219]: https://loopcrack.atlassian.net/browse/KIN-219?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/Controllers/CtrlCategory.php [hunk 1]
```diff
@@ -82,21 +82,26 @@ public function updateCategory($id)
...
-        if ($isDeleted) {
-            $response = [
-                'title'   => 'Eliminación exitosa',
-                'message' => 'Se ha elimnado la categoría correctamente',
-                'type'    => 'success',
-            ];
-        } else {
+        $categoryId = $this->request->getPost('categoryId') ?? '';
+
+        (new CategoryModel())->deleteCategory($categoryId);
```
Comment (Mauriciocr207) [inline app/Controllers/CtrlCategory.php:87]: Here is not neccesary verify if the register was deleted and send a response according to that. Is not neccesary to consider where the delete return `false`?

**Review by Oswaldo-Chan — APPROVED** [review]
👍 approved

**Review by Mauriciocr207 — APPROVED** [review]

**Review by CarlosMay7 — COMMENTED** [review]
It works fine but when we checked the database the products related to the deleted category were still there when those were supposed to be deleted as well. Please check it out

**Comment by Cristianpan** [comment]
> It works fine but when we checked the database the products related to the deleted category were still there when those were supposed to be deleted as well. Please check it out

That is because of how the test categories have been created, if you look at the seeders you can see that there are only 3 files created and those same files are associated to any entity that needs them, then when you delete the category, because of these associations between files, is what "fails" however in a normal flow that wouldn't happen, you can try creating a new category and you will see that everything works normally.

File: app/Models/CategoryModel.php [hunk 2]
```diff
@@ -38,6 +39,23 @@ public function createCategory(array $categoryData)
         }
     }
 
+<<<<<<< HEAD
```
Comment (LUGAMAFE) [inline app/Models/CategoryModel.php:42]: @Cristianpan beware of these errors when resolving merge conflicts
```suggestion

```

File: app/Models/CategoryModel.php [hunk 3]
```diff
@@ -38,6 +39,23 @@ public function createCategory(array $categoryData)
...
+
+            $fileModel    = new FileModel();
+            $imageDeleted = $fileModel->delete($categoryData['categoryImageId']);
+            $iconDeleted  = $fileModel->delete($categoryData['categoryIconId']);
+
+            if (! $iconDeleted || ! $imageDeleted || ! $categoryDeleted) {
+                throw new Exception('No se ha podido eliminar la categoría');
+            }
+
+=======
```
Comment (LUGAMAFE) [inline app/Models/CategoryModel.php:58]: same
```suggestion
```

File: app/Models/CategoryModel.php [hunk 4]
```diff
@@ -60,6 +78,7 @@ public function updateCategory(string $categoryId, array $categoryData)
             }
 
             (new CategoryModel())->update($categoryId, $categoryData);
+>>>>>>> 51ecbe590e18479386d7c3b8d914ab2236ef7dad
```
Comment (LUGAMAFE) [inline app/Models/CategoryModel.php:81]: same
```suggestion
```

**Review by LUGAMAFE — CHANGES_REQUESTED** [review]

File: app/Models/CategoryModel.php (hunk 2 above)
Comment (LUGAMAFE) [inline app/Models/CategoryModel.php:42]: @Cristianpan beware of these errors when resolving merge conflicts, you didn't choose which version to use, right
```suggestion
```

**Review by LUGAMAFE — CHANGES_REQUESTED** [review]

File: app/Models/CategoryModel.php (hunk 2 above)
Comment (Cristianpan) [reply app/Models/CategoryModel.php]: Is right, I didn't see that conflict,  so sorry.

**Review by CheloChidoone — APPROVED** [review]
Good Job @Cristianpan

File: app/Controllers/CtrlCategory.php [hunk 5]
```diff
@@ -132,4 +132,28 @@ public function updateCategory(string $categoryId)
...
+            $categoryId = $this->request->getPost('categoryId') ?? '';
+
+            (new CategoryModel())->deleteCategory($categoryId);
+            $response = [
+                'title'   => 'Eliminación exitosa',
+                'message' => 'La categoría ha sido eliminada correctamente',
+                'type'    => 'success',
+            ];
+
+            return redirect()->to('admin/categorias')->with('response', $response);
```
Comment (CarlosMay7) [inline app/Controllers/CtrlCategory.php:148]: Maybe you can move his line outside of the try catch and everything should work fine

File: app/Controllers/CtrlCategory.php [hunk 6]
```diff
@@ -132,4 +132,28 @@ public function updateCategory(string $categoryId)
...
+        } catch (Throwable $th) {
+            $response = [
+                'title'   => 'Oops! Ha ocurrido un error.',
+                'message' => 'Ha ocurrido un error al eliminar los datos de la categoría, por favor intente nuevamente.',
+                'type'    => 'error',
+            ];
+
+            return redirect()->to('admin/categorias')->with('response', $response);
```
Comment (CarlosMay7) [inline app/Controllers/CtrlCategory.php:156]: Because is the same statement as before

File: app/Models/CategoryModel.php [hunk 7]
```diff
@@ -67,4 +68,27 @@ public function updateCategory(string $categoryId, array $categoryData)
...
+
+            $fileModel    = new FileModel();
+            $imageDeleted = $fileModel->delete($categoryData['categoryImageId']);
+            $iconDeleted  = $fileModel->delete($categoryData['categoryIconId']);
+
+            if (! $iconDeleted || ! $imageDeleted || ! $categoryDeleted) {
+                throw new Exception('No se ha podido eliminar la categoría');
+            }
+
+            $this->db->transComplete();
```
Comment (CarlosMay7) [inline app/Models/CategoryModel.php:87]: Check the suggestion you did about the change from transComplete to transCommit

**Review by CarlosMay7 — CHANGES_REQUESTED** [review]
Hey @Cristianpan, we checked and everything looks good overall, just a few comments. 

Also we have a question, shouldn't we delete the category tags along with the category?

CC: @FernandoJoachin

**Comment by Cristianpan** [comment]
> Hey @Cristianpan, we checked and everything looks good overall, just a few comments.
> 
> Also we have a question, shouldn't we delete the category tags along with the category?
> 
> CC: @FernandoJoachin

I've made the suggested changes, but about deleting category tags, it works correctly for me and I've not made it delete through the code because I'm delegating that action to the Db. I'd like that you to confirm if you are using a proper configuration for the Db (the eginee innodb). Likewise if anyone can confirm whether or not it is working correctly for them, that would be nice.

**Review by FernandoJoachin — APPROVED** [review]

**Review by CarlosMay7 — APPROVED** [review]
Nice work, thank you

**Review by CarlosJCD — APPROVED** [review]
Nice work!, remember to resolve comments before doing merge

**Review by LUGAMAFE — APPROVED** [review]

**Pull request merged**